tests/integration/state_test.py

```diff
@@ -136,6 +136,17 @@ def test_change_middle(self):
         self.cfg['web']['environment'] = {'NEW_VAR': '1'}
         new_containers = self.run_up(self.cfg)
 
+        self.assertEqual(
+            set(c.name_without_project for c in new_containers - old_containers),
+            set(['web_1']),
+        )
+
+    def test_change_middle_always_recreate_deps(self):
+        old_containers = self.run_up(self.cfg, **{"always_recreate_deps": True})
+
+        self.cfg['web']['environment'] = {'NEW_VAR': '1'}
+        new_containers = self.run_up(self.cfg, **{"always_recreate_deps": True})
+
         self.assertEqual(
```

Comment:
Please use the `assert A == B` notation here instead (I know the other tests in this file use the `assertEqual` notation, but they were written before we switched to py.test. We'll transition them as well eventually)